crates/qasm3/src/exporter.rs

```diff
@@ -1186,12 +1190,8 @@ impl<'a> QASM3Builder {
         let duration: f64 = Python::with_gil(|py| match param {
             Param::Float(val) => *val,
             Param::ParameterExpression(p) => {
-                let py_obj = p.bind(py);
-                let py_str = py_obj.str().expect("Failed to call str() on Parameter");
-                let name = py_str
-                    .str()
-                    .expect("Failed to convert PyString to &str")
-                    .to_string();
+                // TODO this should probably use p.try_numeric
```

Comment:
To be addressed before merge?

Reply:
I can take a quick stab at it, I wasn't too worried about it because all the rust qasm3 code is experimental. But it doesn't seem to hard to try.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/14799/commits/359ef6a411cb534ec796c7b1cb3af912a7789115